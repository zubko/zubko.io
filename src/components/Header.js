/**
 * Top common part of the website
 *
 * @flow
 */

import React, { useState } from 'react';
import { StaticQuery, graphql } from 'gatsby';

import { Link, MenuVertical, MenuHorizontal } from '../components';
import { Media, Colors } from '../Theme';
import { rhythm } from '../Typography';

const MENU_MEDIA_SPLIT = 780;
const MediaForVertMenu = Media.lessThanPx(MENU_MEDIA_SPLIT);
const MediaForHorzMenu = Media.greaterThanPx(MENU_MEDIA_SPLIT);

type Props = {
  siteTitle: string,
};

const Header = ({ siteTitle }: Props) => {
  const [isMenuOpened, setMenuOpened] = useState(false);
  return (
    <StaticQuery
      query={graphql`
        query MenuQuery {
          allMenuYaml {
            edges {
              node {
                path
                title
              }
            }
          }
        }
      `}
      render={({ allMenuYaml: { edges: menuEdges } }: Object) => {
        const menuLinks = menuEdges.map(e => e.node);
        return (
          <div css={styles.headerOuter}>
            <div css={styles.headerInner}>
              <Link usePlainStyle to="/" css={styles.homeLink}>
                <h3>{siteTitle}</h3>
              </Link>

              <MenuHorizontal
                css={{ [MediaForVertMenu]: { display: 'none' } }}
                links={menuLinks}
              />
              <MenuVertical
                isHead
                isOpened={isMenuOpened}
                setOpened={setMenuOpened}
                css={{ [MediaForHorzMenu]: { display: 'none' } }}
                links={menuLinks}
              />
            </div>
            <MenuVertical
              isBody
              isOpened={isMenuOpened}
              setOpened={setMenuOpened}
              css={{ [MediaForHorzMenu]: { display: 'none' } }}
              links={menuLinks}
            />
          </div>
        );
      }}
    />
  );
};

const styles = {
  headerOuter: {
    marginBottom: `${rhythm(0.5)}`,
  },
  headerInner: {
    paddingTop: rhythm(0.5),
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  homeLink: {
    textDecoration: `none`,
    ':hover': {
      textDecoration: `none`,
    },
    background: 'none',
    textShadow: 'none',
    '> h3': {
      marginTop: 0,
      marginBottom: 0,
      color: Colors.mainDarker,
    },
  },
};

Header.defaultProps = {
  siteTitle: ``,
};

export default Header;
