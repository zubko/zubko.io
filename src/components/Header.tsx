/**
 * Top common part of the website
 */

import { graphql, StaticQuery } from "gatsby";
import { useState } from "react";

import { Link, MenuHorizontal, MenuVertical } from "../components";
import { Colors, Media } from "../Theme";
import { rhythm } from "../Typography";

const MENU_MEDIA_SPLIT = 780;
const MediaForVertMenu = Media.lessThanPx(MENU_MEDIA_SPLIT);
const MediaForHorzMenu = Media.greaterThanPx(MENU_MEDIA_SPLIT);

type Props = {
  siteTitle?: string;
};

const Header = ({ siteTitle = "" }: Props) => {
  const [isMenuOpened, setMenuOpened] = useState(false);
  return (
    <StaticQuery<Queries.MenuQueryQuery>
      query={graphql`
        query MenuQuery {
          allMenuYaml {
            edges {
              node {
                path
                title
              }
            }
          }
        }
      `}
      render={({ allMenuYaml: { edges: menuEdges } }) => {
        const menuLinks = menuEdges.map(e => e.node);
        return (
          <div css={styles.headerOuter}>
            <div css={styles.headerInner}>
              <Link usePlainStyle to="/" css={styles.homeLink}>
                <h1>{siteTitle}</h1>
              </Link>

              <MenuHorizontal
                css={{ [MediaForVertMenu]: { display: "none" } }}
                links={menuLinks}
              />
              <MenuVertical
                isHead
                isOpened={isMenuOpened}
                setOpened={setMenuOpened}
                css={{ [MediaForHorzMenu]: { display: "none" } }}
                links={menuLinks}
              />
            </div>
            <MenuVertical
              isBody
              isOpened={isMenuOpened}
              setOpened={setMenuOpened}
              css={{ [MediaForHorzMenu]: { display: "none" } }}
              links={menuLinks}
            />
          </div>
        );
      }}
    />
  );
};

const styles = {
  headerOuter: {
    marginBottom: `${rhythm(0.5)}`
  },
  headerInner: {
    paddingTop: rhythm(0.5),
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center"
  },
  homeLink: {
    textDecoration: `none`,
    ":hover": {
      textDecoration: `none`
    },
    background: "none",
    textShadow: "none",
    "> h1": {
      color: Colors.mainDarker,
      margin: 0,
      fontWeight: 600,
      fontSize: "1.21225rem" // h3
    }
  }
};

export default Header;
