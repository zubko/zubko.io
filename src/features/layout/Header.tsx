/**
 * Top common part of the website
 */

import { graphql, StaticQuery } from "gatsby";
import { useState } from "react";
import styled from "styled-components";

import { Link } from "../../components/Link";
import { Colors, Media } from "../../Theme";
import { rhythm } from "../../Typography";

import { MenuHorizontal as MenuHorizontalOrig } from "./MenuHorizontal";
import { MenuVertical as MenuVerticalOrig } from "./MenuVertical";

const MENU_MEDIA_SPLIT = 780;
const MediaForVertMenu = Media.lessThanPx(MENU_MEDIA_SPLIT);
const MediaForHorzMenu = Media.greaterThanPx(MENU_MEDIA_SPLIT);

type Props = {
  siteTitle?: string;
};

type LinkNode = {
  path: string;
  title: string;
};

export const Header = ({ siteTitle = "" }: Props) => {
  const [isMenuOpened, setMenuOpened] = useState(false);
  return (
    <StaticQuery<Queries.HeaderMenuQuery>
      query={graphql`
        query HeaderMenu {
          allMenuYaml {
            edges {
              node {
                path
                title
              }
            }
          }
        }
      `}
      render={({ allMenuYaml: { edges: menuEdges } }) => {
        const menuLinks = menuEdges.map((e) => e.node as LinkNode);
        return (
          <HeaderOuter>
            <HeaderInner>
              <HomeLink usePlainStyle to="/">
                <h1>{siteTitle}</h1>
              </HomeLink>

              <MenuHorizontal links={menuLinks} />
              <MenuVertical
                isHead
                isOpened={isMenuOpened}
                onOpened={setMenuOpened}
                links={menuLinks}
              />
            </HeaderInner>
            <MenuVertical
              isBody
              isOpened={isMenuOpened}
              onOpened={setMenuOpened}
              links={menuLinks}
            />
          </HeaderOuter>
        );
      }}
    />
  );
};

const HeaderOuter = styled.div`
  margin-bottom: ${rhythm(0.5)};
`;

const HeaderInner = styled.div`
  padding-top: ${rhythm(0.5)};
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
`;

const HomeLink = styled(Link)`
  text-decoration: none;
  :hover {
    text-decoration: none;
  }
  background: none;
  text-shadow: none;
  > h1 {
    color: ${Colors.mainDarker};
    margin: 0;
    font-weight: 600;
    font-size: 1.21225rem; // h3
  }
`;

const MenuHorizontal = styled(MenuHorizontalOrig)`
  ${MediaForVertMenu} {
    display: none;
  }
`;

const MenuVertical = styled(MenuVerticalOrig)`
  ${MediaForHorzMenu} {
    display: none;
  }
`;
