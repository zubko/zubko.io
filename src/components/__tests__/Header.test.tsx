import React from "react";
import renderer from "react-test-renderer";
import { StaticQuery } from "gatsby";

import Header from "../Header";

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      allMenuYaml: {
        edges: [
          {
            node: {
              path: "/path-1/",
              title: "title-1",
            },
          },
          {
            node: {
              path: "/path-2/",
              title: "title-2",
            },
          },
        ],
      },
    })
  );
});

describe("Header", () => {
  it("renders without regressions", () => {
    const tree = renderer.create(<Header siteTitle="siteTitle" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
