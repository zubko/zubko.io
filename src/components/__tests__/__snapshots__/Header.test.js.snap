// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without regressions 1`] = `
<div
  css={
    Object {
      "marginBottom": "0.809rem",
    }
  }
>
  <div
    css={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingTop": "0.809rem",
      }
    }
  >
    <a
      css={
        Object {
          ":hover": Object {
            "textDecoration": "none",
          },
          "> h3": Object {
            "color": "#2C8127",
            "marginBottom": 0,
            "marginTop": 0,
          },
          "background": "none",
          "textDecoration": "none",
          "textShadow": "none",
        }
      }
      href="/"
    >
      <h3>
        siteTitle
      </h3>
    </a>
    <div
      css={
        Object {
          "@media (max-width: 779px)": Object {
            "display": "none",
          },
        }
      }
    >
      <a
        css={
          Object {
            ":hover": Object {},
            "background": "none",
            "fontSize": "1rem",
            "fontWeight": "medium",
            "textDecoration": "none",
          }
        }
        href="/path-1/"
      >
        title-1
      </a>
      <span
        css={
          Object {
            "color": "white",
            "cursor": "default",
            "marginLeft": "0.5rem",
            "marginRight": "0.5rem",
          }
        }
      >
         
      </span>
      <a
        css={
          Object {
            ":hover": Object {},
            "background": "none",
            "fontSize": "1rem",
            "fontWeight": "medium",
            "textDecoration": "none",
          }
        }
        href="/path-2/"
      >
        title-2
      </a>
    </div>
    <div
      css={
        Object {
          "@media (min-width: 780px)": Object {
            "display": "none",
          },
        }
      }
    >
      <button
        css={
          Array [
            Object {
              ":focus": Object {
                "outline": 0,
              },
              "backgroundColor": "transparent",
              "border": "1px solid #2C8127",
              "color": "#2C8127",
              "cursor": "pointer",
              "padding": "0.5rem",
            },
            null,
          ]
        }
        onClick={[Function]}
      >
        Menu
      </button>
    </div>
  </div>
  <div
    css={
      Object {
        "@media (min-width: 780px)": Object {
          "display": "none",
        },
      }
    }
  >
    <div
      css={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "column",
            "paddingBottom": "0.809rem",
            "paddingTop": "1.618rem",
          },
          Object {
            "display": "none",
          },
        ]
      }
    >
      <a
        css={
          Object {
            ":hover": Object {},
            "background": "none",
            "fontSize": "1rem",
            "fontWeight": "medium",
            "textDecoration": "none",
          }
        }
        href="/path-1/"
      >
        title-1
      </a>
      <span
        css={
          Object {
            "color": "white",
            "cursor": "default",
            "height": "0.5rem",
          }
        }
      >
         
      </span>
      <a
        css={
          Object {
            ":hover": Object {},
            "background": "none",
            "fontSize": "1rem",
            "fontWeight": "medium",
            "textDecoration": "none",
          }
        }
        href="/path-2/"
      >
        title-2
      </a>
    </div>
  </div>
</div>
`;
